app/controllers/employment.py

```diff
@@ -476,3 +477,35 @@ def mutate(cls, _, info, employment_id):
             db.session.commit()
 
         return Void(success=True)
+
+
+class ChangeEmployeeRole(AuthenticatedMutation):
+    class Arguments:
+        employment_id = graphene.Argument(
+            graphene.Int,
+            required=True,
+            description="Identifiant du rattachement pour lequel le rôle doit être changé",
+        )
+        has_admin_rights = graphene.Argument(
+            graphene.Boolean,
+            required=True,
+            description="Précise si le salarié rattaché est gestionnaire de l'entreprise, et s'il pourra donc avoir les droits de consultation et d'admnistration associés.",
+        )
+
+    Output = EmploymentOutput
+
+    @classmethod
+    @with_authorization_policy(
+        company_admin,
+        get_target_from_args=lambda *args, **kwargs: Employment.query.get(
+            kwargs["employment_id"]
+        ).company_id,
+        error_message="Actor is not authorized to change employee role",
+    )
+    def mutate(cls, _, info, employment_id, has_admin_rights):
```

Comment:
Shouldn't we check if the user is an admin of the company of the employment ?

Reply:
This is already done with the _company_admin_ authorization policy.